cmd/main.go

```diff
@@ -44,27 +46,47 @@ func main() {
 		runtime.GOMAXPROCS(runtime.NumCPU())
 	}
 
-	const tlsCertRoot = "/etc/pki/tls/certs"
-	const runtimeCertRoot = "/etc/docker/certs.d"
-
-	clusterCASrc := fmt.Sprintf("%s/ca.crt", builder.SecretCertsMountPath)
-	clusterCADst := fmt.Sprintf("%s/cluster.crt", tlsCertRoot)
-	fs := s2ifs.NewFileSystem()
-	err := fs.Copy(clusterCASrc, clusterCADst, map[string]string{})
-	if err != nil {
-		fmt.Printf("Error setting up cluster CA cert: %v", err)
-		os.Exit(1)
-	}
-
-	runtimeCASrc := fmt.Sprintf("%s/certs.d", builder.ConfigMapCertsMountPath)
-	err = fs.CopyContents(runtimeCASrc, runtimeCertRoot, map[string]string{})
-	if err != nil {
-		fmt.Printf("Error setting up service CA cert: %v", err)
-		os.Exit(1)
+	if !inUserNamespace() {
```

Comment:
So why do we only do this when not in user namespaces?

How is this setup provided when we are in user namespaces?

Looking for a comment that answers those questions.  Thanks.

Reply:
If we've set up a user namespace, then we know that we weren't started as UID 0 in the container, so these locations are not writable.  If we try to write to those locations anyway, we hit a permissions error and exit.  It's an unsolved problem for cases where we start the container as a UID other than 0.

Reply:
OK caputure that in a comment in the code and I'm good on this one

Reply:
So as non-root we can't write to `/etc/docker/certs.d`? Interesting, may need to determine how we can have the build controller mount these for us.

Reply:
Will this also impact our entrypoint script that generates the trust bundle?

Reply:
Mounting the `certs.d` subpath of the `build-ca-bundles` volume to `/etc/docker/certs.d` _should_ work, since the directory isn't even in the builder image to begin with.  The `cluster.crt` that we drop in `/etc/pki/tls/certs` is trickier, since the directory already includes content that we want to still have there.  We should be able to mount the file itself from the volume that contains the service account secrets, but I haven't personally tried that.
This may all be moot, though, since I think ensuring the directories exist and setting them group-writeable in the image will also work, with no code changes.

Reply:
Going to try creating the directories and setting them group-writable in the `Dockerfile` instead.  That'll probably work better for the entrypoint script, too.

Reply:
OK if !inUserNamespace() condition has been removed along with the Dockerfile changes @nalind mentions ^^